feat: Add theme solution to support multiple theme switching

gene9831 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - ThemeProvider with theme & color-mode controls, persistence, nesting and a useTheme hook; multiple theme demos added.
  - Prompts gain a size option (small/medium/large) with a default and improved badge rendering.

- **Style**
  - Expanded design tokens and variable-driven Prompt/Prompts styles, centralized spacing, conditional horizontal scrolling, demo footer icon and theme demo CSS tweaks.

- **Documentation**
  - New Theme guide, expanded Prompts docs (size, CSS variables) and added "Theme 主题" sidebar entry.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gene9831 committed:
feat: add ThemeProvider component and related functionality for theme management

- Introduced ThemeProvider component to manage light and dark themes.
- Added constants and types for theme handling.
- Implemented useTheme composable for theme toggling and storage management.
- Updated index files to export new ThemeProvider and related utilities.

gene9831 committed:
feat: implement CSS variable design for prompts component

- Introduced a three-layer variable system for prompts: theme layer, component layer, and style layer.
- Added size property to PromptProps for controlling prompt sizes (small, medium, large).
- Updated prompt.vue to utilize size classes and CSS variables for styling.
- Created new demo for size variations of prompts.
- Enhanced documentation to include size property usage and examples.

gene9831 committed:
feat: implement theme management and color mode functionality

- Added ColorModeSubject for managing light and dark themes.
- Introduced ThemeProvider component to wrap applications for theme and color mode management.
- Implemented setupDarkModeListener to handle dark mode changes.
- Created demo components for theme and color mode switching.
- Enhanced documentation for ThemeProvider usage and examples.
- Refactored styles to support theme and color mode attributes.

gene9831 committed:
refactor: consolidate theme styles into dedicated components

- Moved theme-related CSS variables from individual components to centralized styles in Storage.vue and Theme.vue.
- Updated StorageComp.vue and ThemeComp.vue to remove redundant style definitions.
- Enhanced scoped styles for better theme management across light and dark modes.

gene9831 committed:
docs: expand ThemeProvider documentation with API details and usage examples

- Added comprehensive API section for ColorMode, ThemeStorage, and ThemeProviderProps.
- Included details on the useTheme composable and its return values.
- Enhanced examples to illustrate theme and color mode management.

gene9831 committed:
fix: improve theme detection and event handling in ThemeProvider

- Moved media query initialization for dark mode inside a window check to prevent errors in server-side rendering.
- Added event listener for dark mode changes and ensured proper cleanup on component unmount.
- Added a safeguard to prevent document queries when not in a browser environment.

gene9831 committed:
docs: clarify ThemeProvider usage in documentation

- Updated the description of the `target-element` prop in the ThemeProvider documentation for better clarity.
- Enhanced the explanation of how CSS variables are applied through the ThemeProvider.

gene9831 committed:
fix: ensure proper subscription management for ColorModeSubject in Layout component

- Updated the subscription handling for ColorModeSubject to prevent potential memory leaks by using an optional chaining operator for unsubscribe.
- Improved the readability of the onMounted and onUnmounted lifecycle hooks in the Layout component.

gene9831 committed:
refactor: standardize CSS variable naming and improve theme styles

- Renamed CSS variables for prompts and themes to follow a consistent naming convention.
- Updated styles in Storage.vue, Theme.vue, and prompt.vue to utilize the new variable names.
- Enhanced the theme management system by consolidating styles and improving readability across components.

gene9831 committed:
refactor: update CSS variable names and enhance prompt styles

- Renamed shadow elevation variable for consistency in naming conventions.
- Added a comment to improve specificity of the CSS selector for theme management in prompt styles.

gene9831 committed:
Merge branch 'develop' of github.com:opentiny/tiny-robot into feat/theme-provider

gene9831 committed:
refactor: enhance footer component and update refresh icon

- Replaced text with an SVG icon for the refresh action in the footer.
- Improved footer styling for better alignment and visual consistency.
- Updated the refresh SVG to use currentColor for better adaptability in theming.

gene9831 committed:
feat: update theme documentation and add nested theme demo

- Renamed 'ThemeProvider' to 'Theme' in the configuration for clarity.
- Introduced a new demo for nested theme usage in `NestedTheme.vue`.
- Added comprehensive documentation for the theme system, including usage, settings, and API details.

gene9831 committed:
refactor: remove unused CSS variables for control components

- Deleted obsolete CSS variables related to control component backgrounds to streamline the stylesheet.
- Improved overall maintainability of the variables.css file by focusing on relevant styles.

gene9831 committed:
refactor: update CSS variable names and improve styling consistency

- Renamed CSS variables for text and spacing to enhance clarity and maintainability.
- Adjusted footer component styles to use updated variable names for better theming support.
- Streamlined prompt styles by aligning color variables with the new naming convention.

gene9831 committed:
Merge branch 'develop' of github.com:opentiny/tiny-robot into feat/theme-provider

gene9831 committed:
feat: enhance prompt component with new badge feature and refactor styles

- Added a 'NEW' badge to small, medium, and large prompt items for better visibility.
- Refactored CSS variables in prompt styles to simplify padding and badge handling.
- Replaced the old prompt.css with a new prompt.less file for improved styling management.
- Introduced prompts.less for additional prompt-related styles.

gene9831 committed:
docs: update prompts and theme documentation with CSS variable details

- Revised prompts documentation to include comprehensive CSS variable definitions for prompt components, enhancing customization options.
- Clarified theme documentation regarding the use of CSS variables and updated prop descriptions for `ThemeProvider` and `useTheme` hooks.

gene9831 committed:
Merge branch 'develop' of github.com:opentiny/tiny-robot into feat/theme-provider

gene9831 committed:
docs: update prompts documentation to streamline size variable definitions

- Simplified the description of prompt component size variables by consolidating them into a single table format.
- Clarified the relationship between size options (`small`, `medium`, `large`) and their corresponding CSS variable names.

gene9831 committed:
refactor: improve dark mode listener and update prompt component styles

- Ensured only one dark mode listener is registered to prevent multiple subscriptions.
- Updated prompt component to use `import type` for better TypeScript practices.
- Simplified CSS properties in prompt styles by replacing `background-color` with `background` for consistency.

gene9831 committed:
Merge branch 'develop' of github.com:opentiny/tiny-robot into feat/theme-provider

gene9831 committed:
Merge branch 'develop' of github.com:opentiny/tiny-robot into feat/theme-provider

gene9831 committed:
refactor: streamline prompt component CSS variables

- Consolidated CSS variable definitions for the prompt component by removing redundant selectors.
- Improved organization of variable declarations for better readability and maintainability.

gene9831 committed:
refactor: enhance CSS variable definitions for prompt components

- Added support for theme and color mode attributes in CSS variable declarations.
- Improved consistency in variable definitions across prompt and prompts styles.

gene9831 committed:
refactor: simplify CSS variable selectors for prompt components

- Removed redundant selectors for theme and color mode in prompt and prompts styles.
- Enhanced clarity and maintainability of CSS variable definitions.

gene9831 committed:
feat: enhance theme provider with system color mode support

- Added `systemColorMode` to the `useTheme` hook and provided it in the `ThemeProvider`.
- Updated documentation to reflect the usage of `systemColorMode`.
- Introduced a new constant for `SYSTEM_COLOR_MODE` in the theme provider constants.

gene9831 committed:
fix: improve theme data handling in ThemeProvider

- Enhanced the logic for retrieving theme and color mode from storage.
- Added validation to ensure only valid color modes ('auto', 'light', 'dark') are set.

SonyLeo commented on packages/components/src/theme-provider/useTheme.ts:
```diff
@@ -0,0 +1,69 @@
+import type { ColorMode } from './index.type'
+
+const WARN_MESSAGE =
+  'Theme context not available, cannot set theme data. Condisder using ThemeProvider to wrap your app.'
```
Condisder 是否存在拼写错误？

SonyLeo commented on packages/components/src/theme-provider/index.vue:
```diff
@@ -0,0 +1,106 @@
+    if (!targetElement) return
+
+    targetElement.setAttribute(THEME_ATTR_NAME, innerTheme.value)
+    targetElement.setAttribute(COLOR_MODE_ATTR_NAME, resolvedColorMode.value)
```
setAttribute() 需要 string 类型， resolvedColorMode.value 可能为 `undefined`

建议外部做非空校验，如

if(resolvedColorMode.value) {
   // 其余代码
}

gene9831 committed:
fix: correct typo in warning message for theme context

- Fixed a typo in the warning message regarding the usage of ThemeProvider in the theme context.
